# 140+ Basic Python Programs
### Questions Only — No Answers

## Program 1
Write a Python program to print "Hello Python".

In [1]:
# Write your code here
print("Hello")

Hello


## Program 2
Write a Python program to do arithmetical operations addition and division.

In [3]:
# Write your code here
a = int(input("Emter an number1:"))
b = int(input("Enter an number2:"))
divide = a / b
add = a + b 
print(f"Result is :",add)
print(f"Division is :" , divide)

Result is : 145
Division is : 1.5892857142857142


## Program 3
Write a Python program to find the area of a triangle.

In [5]:
# Write your code here
base =  float(input("Enter the base:"))
height = float(input("Enter the height:"))
area = 0.5*height*base
print(area)

15.0


## Program 4
Write a Python program to swap two variables.

In [6]:
# Write your code here
a =  12 
b =  14

a = b 
print(a)
print(b)



14
14


## Program 5
Write a Python program to generate a random number.

In [36]:
# Write your code here
import random

a = random.random()
print(a)

0.4724584819700479


## Program 6
Write a Python program to convert kilometers to miles.

In [39]:
# Write your code her
a =  int(input("Enter Km to convert: "))
b = a * 0.621371
print(f"Km to Miles is :",b)

Km to Miles is : 34.796776


## Program 7
Write a Python program to convert Celsius to Fahrenheit.

In [42]:
# Write your code here
a = float(input("Enter Temp in Celsius"))
b = a * 9/5 + 32
print(f"Celsius to Fahrenheit is: ",b,"°F")

Celsius to Fahrenheit is:  132.8 °F


## Program 8
Write a Python program to display calendar.

In [43]:
# Write your code here
import calendar
year = int(input("Enter year"))
month = int(input("enter month"))
cal = calendar.month(year, month)
print(cal)

   November 2000
Mo Tu We Th Fr Sa Su
       1  2  3  4  5
 6  7  8  9 10 11 12
13 14 15 16 17 18 19
20 21 22 23 24 25 26
27 28 29 30



## Program 9
Write a Python program to solve quadratic equation.

The standard form of a quadratic equation is: ax² + bx + c = 0

where a, b and c are real numbers and a ≠ 0

The solutions of this quadratic equation is given by: (−b ± (b² − 4ac)^(1/2)) / (2a)

In [ ]:
# Write your code here
import cmath
a =  float(input('Enter an number1'))
b =  float(input("Enter an number2"))
c =  float(input("Enter an number3"))

if a ==0:
    print("this is not quadratic equation:")
else:
    d =  b**2 - 4*a*c

    sol1 = (-b + cmath.sqrt(d)) / (2*a)
    sol2 = (b - cmath.sqrt(d)) / (2*a)

    print("X1 = ",sol1)
    print("X2 = ",sol2)
    


X1 =  (2.618033988749895+0j)
X2 =  (-2.618033988749895-0j)


## Program 10
Write a Python program to swap two variables without temp variable.

In [56]:
# Write your code here
a = 12 
b = 13
a,b = b,a
print(a)
print(b)

13
12


## Program 11
Write a Python Program to Check if a Number is Positive, Negative or Zero.

In [59]:
a =  int(input("Enter Num:"))
if a > 0:
    print("Number is positive")
elif a < 0:
    print("Number is negative")
elif a == 0:
    print("num is zero")

Number is negative


## Program 12
Write a Python Program to Check if a Number is Odd or Even.

In [61]:
# Write your code here
a =  int(input("enter an number: "))
if a % 2 == 0:
    print("Number is Even")
else:
    print("Number is odd")


Number is odd


## Program 13
Write a Python Program to Check Leap Year.

In [64]:
# Write your code herea
a = int(input("Enter a year: "))
if a % 4 == 0:
    print("year is leap")
else:
    print("Year is not Leap")

year is leap


## Program 14
Write a Python Program to Check Prime Number.

**Prime Numbers:** A prime number is a whole number that cannot be evenly divided by any other number except for 1 and itself. For example, 2, 3, 5, 7, 11, and 13 are prime numbers.

In [70]:
# Write your code here
a =  int(input("Enter an number:"))
flag = False
if a == 1:
    print("Num is not Prime")
elif  a > 1:
    for i in range(2,a):
        if a % i == 0:
            flag = True
        break
if flag:
    print(f"{a}is not prime")
else:
    print(f"{a} is prime")

Num is not Prime
1 is prime


## Program 15
Write a Python Program to Print all Prime Numbers in an Interval of 1-10.

In [ ]:
# Write your code here
# Using list comprehension
a = [n for n in range(2,11)if all(n%i !=0 for i in range(2,int(n**0.5)+1))]
print("Prime numbers from 1 to 10:",a)

Prime numbers from 1 to 10: [2, 3, 5, 7]


## Program 16
Write a Python Program to Find the Factorial of a Number.

In [ ]:
# Write your code here
a = int(input("Enter an Number:-"))
fact = 1
if a < 0:
    print("factorial not possible")
elif a == 0:
    print("Factoprial of 0 is 1")
else:
    for i in range(1,a+1):
        fact = fact*i
    print(f"Factorial of {a} is ",fact)

## Program 17
Write a Python Program to Display the multiplication Table.

In [10]:
# Write your code here
num =  int(input("Enter an number : "))

for i in range(1,11):
    print(f"{num} X {i}",num*i)

14 X 1 14
14 X 2 28
14 X 3 42
14 X 4 56
14 X 5 70
14 X 6 84
14 X 7 98
14 X 8 112
14 X 9 126
14 X 10 140



## Program 18
Write a Python Program to Print the Fibonacci sequence.

**Fibonacci sequence:** A series of numbers where each number is the sum of the two preceding ones, starting with 0 and 1.

0, 1, 1, 2, 3, 5, 8, 13, 21, 34, ...

F(0) = 0, F(1) = 1, F(n) = F(n-1) + F(n-2) for n > 1

In [ ]:
# Write your code here

def fib_rec(n):
    if n <= 1:
        return n
    return fib_rec(n-1) + fib_rec(n-2)
print("first 10 fibonacci Sequence are:")
for i in range(11):
    print(fib_rec(i),end=" ")

first 10 fibonacci Sequence are:
0 1 1 2 3 5 8 13 21 34 55 

## Program 19
Write a Python Program to Check Armstrong Number?

**Armstrong Number:** A number equal to the sum of its own digits, each raised to a power equal to the number of digits.

For example, 153: 1³ + 5³ + 3³ = 1 + 125 + 27 = 153

In [ ]:
num  = int(input("Enter an Num"))

order = len(str(num))
sum = 0
temp = num 
while temp > 0:
    digit = temp % 10
    sum += digit ** order
    temp //= 10
if num == sum:
    print('Armstring num')

## Program 20
Write a Python Program to Find Armstrong Number in an Interval.

In [26]:
# Write your code hereL = l = int
l = int(input("enter an lower limit "))
u = int(input("enter an upper limit"))
for num in range(l,u+1):
    order =  len(str(num))
    temp_num = num
    sum = 0
    while temp_num > 0:
        digit = temp_num % 10
        sum += digit ** order
        temp_num //= 10
    if num == sum:
        print(num)


153
370
371
407


## Program 21
Write a Python Program to Find the Sum of Natural Numbers.

**Natural numbers:** N = 1, 2, 3, 4, 5, 6, 7, 8, ...

In [28]:
# Write your code here
a =  int(input("enter an number"))
sum  = 0 
for i in range(1,a+1):
    sum += i
print(sum)

5050


## Program 22
Write a Python Program to Find LCM.

**Least Common Multiple (LCM):** The smallest multiple that is exactly divisible by two or more numbers.

Formula: LCM(a, b) = |a · b| / GCD(a, b)

In [1]:
# Write your code here
import math 
a = int(input("Enter a number"))
b = int(input("Enter an second number"))
lcm =  abs(a*b) // math.gcd(a,b) 
print(f"LCM IS : ",lcm)

LCM IS :  60


## Program 23
Write a Python Program to Find HCF.

**Highest Common Factor (HCF):** The largest positive integer that divides two or more numbers without leaving a remainder.

Formula: HCF(a, b) = GCD(a, b)

In [2]:
# Write your code here
a =  int(input("Enter first Num"))
b = int(input("Enter an second no"))

hcf  =  math.gcd(a,b)
print(f"HCF is : ",hcf)

HCF is :  1


## Program 24
Write a Python Program to Convert Decimal to Binary, Octal and Hexadecimal.

In [14]:
# Write your code here
#  decimal to binary 

a =  int(input("Enter an number:"))
print(f"Decimal to Binary is :",bin(a)[2:])
print(f"Octal to Binary is: ",hex(a))

Decimal to Binary is : 10000
Octal to Binary is:  0x10


## Program 25
Write a Python Program To Find ASCII value of a character.

**ASCII:** American Standard Code for Information Interchange — each character has a unique numeric value.

In [3]:
# Write your code here)
a =  str(input("Enter charcters:"))
print(f"ASCII Value of {a} is : ",ord(a))
         

ASCII Value of j is :  106


## Program 26
Write a Python Program to Make a Simple Calculator with 4 basic mathematical operations (addition, subtraction, multiplication, division).

In [1]:
# Write your code here
def add(a,b):
    return a + b 
def sub(a,b):
    return a - b
def mul(a,b):
    return a * b
def div(a,b):
    return a / b

menu = """
1. For addition
2. for Subtraction
3. for multiplication
4. for division

"""
while True:
    choice = input("Enter an Choice 1/2/3/4: ")

    if choice in ('1','2','3','4'):
        try:
            a = float(input("Enter an Number:"))
            b = float(input("Enter an second number:"))
        except ValueError:
            print("Invalid input plz enter valid values")
            continue

        if choice == '1':
            print(a, '+', b,':',add(a,b))
        elif choice == '2':
            print(a, '-',b,':',sub(a,b))
        elif choice == '3':
            print(a,'x',b,':',mul(a,b))
        elif choice == '4':
            print(a,'/',b,':',div(a,b))
        next_cal = input("You Want to do next calculation? (Yes/No)")
        if next_cal == 'No':
            break
    else:
        print("invalid input")


20.0 - 10.0 : 10.0
56.0 - 89.0 : -33.0
20.0 x 10.0 : 200.0
20.0 / 10.0 : 2.0


## Program 27
Write a Python Program to Display Fibonacci Sequence Using Recursion.

In [3]:
# Write your code here
def fibo(i):
    if i <= 1:
        return i
    else:
        return (fibo(i - 1) + fibo(i -2))
num = int(input("Enter an number"))
if num <= 0:
    print("plz Enter a positive Number:")
else:
    print("Fibonacchi series:",end=" ")
    for i in range(num):
        print(fibo(i),end=' ')

Fibonacchi series: 0 1 1 2 3 5 

## Program 28
Write a Python Program to Find Factorial of Number Using Recursion.

n! = n × (n-1) × (n-2) × ... × 3 × 2 × 1

For example: 5! = 5 × 4 × 3 × 2 × 1 = 120

In [ ]:
# Write your code here


## Program 29
Write a Python Program to calculate your Body Mass Index (BMI).

Formula: BMI = Weight (kg) / Height (m)²

Alternatively (imperial): BMI = (Weight (lb) / Height (in)²) × 703

In [ ]:
# Write your code here


## Program 30
Write a Python Program to calculate the natural logarithm of any number.

The natural logarithm ln(x) is the logarithm to the base e (≈ 2.71828).

In [ ]:
# Write your code here


## Program 31
Write a Python Program for cube sum of first n natural numbers.

In [ ]:
# Write your code here


## Program 32
Write a Python Program to find sum of array.

In [ ]:
# Write your code here


## Program 33
Write a Python Program to find largest element in an array.

In [ ]:
# Write your code here


## Program 34
Write a Python Program for array rotation.

In [ ]:
# Write your code here


## Program 35
Write a Python Program to Split the array and add the first part to the end.

In [ ]:
# Write your code here


## Program 36
Write a Python Program to check if given array is Monotonic.

A monotonic array is one that is entirely non-increasing or non-decreasing.

In [ ]:
# Write your code here


## Program 37
Write a Python Program to Add Two Matrices.

In [ ]:
# Write your code here


## Program 38
Write a Python Program to Multiply Two Matrices.

In [ ]:
# Write your code here


## Program 39
Write a Python Program to Transpose a Matrix.

In [ ]:
# Write your code here


## Program 40
Write a Python Program to Sort Words in Alphabetic Order.

In [ ]:
# Write your code here


## Program 41
Write a Python Program to Remove Punctuation From a String.

In [ ]:
# Write your code here


## Program 42
Write a Python program to find sum of elements in a list.

In [ ]:
# Write your code here


## Program 43
Write a Python program to check if the given number is a Disarium Number.

**Disarium Number:** A number equal to the sum of its digits each raised to the power of its respective position.

For example, 89 is a Disarium number because 8¹ + 9² = 8 + 81 = 89.

In [ ]:
# Write your code here


## Program 44
Write a Python program to print all disarium numbers between 1 to 100.

In [ ]:
# Write your code here


## Program 45
Write a Python program to check if the given number is Happy Number.

**Happy Number:** A positive integer where, repeatedly replacing the number by the sum of squares of its digits, eventually reaches 1.

For example, 19 is a Happy Number:
- 1² + 9² = 82
- 8² + 2² = 68
- 6² + 8² = 100
- 1² + 0² + 0² = 1

In [ ]:
# Write your code here


## Program 46
Write a Python program to print all happy numbers between 1 and 100.

In [ ]:
# Write your code here


## Program 47
Write a Python program to determine whether the given number is a Harshad Number.

**Harshad Number:** An integer divisible by the sum of its digits.

For example: 18 is a Harshad number because 1 + 8 = 9, and 18 is divisible by 9.

In [ ]:
# Write your code here


## Program 48
Write a Python program to print all pronic numbers between 1 and 100.

**Pronic Number:** The product of two consecutive integers n and (n+1). Formula: Pₙ = n × (n+1)

First few: P₁ = 2, P₂ = 6, P₃ = 12, P₄ = 20

In [ ]:
# Write your code here


## Program 49
Write a Python program to find sum of elements in a list.

In [ ]:
# Write your code here


## Program 50
Write a Python program to Multiply all numbers in the list.

In [ ]:
# Write your code here


## Program 51
Write a Python program to find smallest number in a list.

In [ ]:
# Write your code here


## Program 52
Write a Python program to find largest number in a list.

In [ ]:
# Write your code here


## Program 53
Write a Python program to find second largest number in a list.

In [ ]:
# Write your code here


## Program 54
Write a Python program to find N largest elements from a list.

In [ ]:
# Write your code here


## Program 55
Write a Python program to print even numbers in a list.

In [ ]:
# Write your code here


## Program 56
Write a Python program to print odd numbers in a List.

In [ ]:
# Write your code here


## Program 57
Write a Python program to Remove empty List from List.

In [ ]:
# Write your code here


## Program 58
Write a Python program to Cloning or Copying a list.

In [ ]:
# Write your code here


## Program 59
Write a Python program to Count occurrences of an element in a list.

In [ ]:
# Write your code here


## Program 60
Write a Python program to find words which are greater than given length k.

In [ ]:
# Write your code here


## Program 61
Write a Python program for removing i-th character from a string.

In [ ]:
# Write your code here


## Program 62
Write a Python program to split and join a string.

In [ ]:
# Write your code here


## Program 63
Write a Python program to check if a given string is binary string or not.

In [ ]:
# Write your code here


## Program 64
Write a Python program to find uncommon words from two Strings.

In [ ]:
# Write your code here


## Program 65
Write a Python program to find all duplicate characters in string.

In [ ]:
# Write your code here


## Program 66
Write a Python Program to check if a string contains any special character.

In [ ]:
# Write your code here


## Program 67
Write a Python program to Extract Unique dictionary values.

In [ ]:
# Write your code here


## Program 68
Write a Python program to find the sum of all items in a dictionary.

In [ ]:
# Write your code here


## Program 69
Write a Python program to Merge two Dictionaries.

In [ ]:
# Write your code here


## Program 70
Write a Python program to convert key-values list to flat dictionary.

In [ ]:
# Write your code here


## Program 71
Write a Python program to insertion at the beginning in OrderedDict.

In [ ]:
# Write your code here


## Program 72
Write a Python program to check order of character in string using OrderedDict().

In [ ]:
# Write your code here


## Program 73
Write a Python program to sort Python Dictionaries by Key or Value.

In [ ]:
# Write your code here


## Program 74
Write a program that calculates and prints the value according to the given formula:

Q = Square root of (2CD/H)

Following are the fixed values of C and H: C is 50, H is 30.

D is the variable whose values should be input to your program in a comma-separated sequence.

**Example:**
- Input: 100,150,180
- Output: 18,22,24

In [ ]:
# Write your code here


## Program 75
Write a program which takes 2 digits, X, Y as input and generates a 2-dimensional array. The element value in the i-th row and j-th column should be i*j.

Note: i=0,1,..,X-1; j=0,1,..,Y-1.

**Example:**
- Input: 3,5
- Output: [[0, 0, 0, 0, 0], [0, 1, 2, 3, 4], [0, 2, 4, 6, 8]]

In [ ]:
# Write your code here


## Program 76
Write a program that accepts a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

**Example:**
- Input: without,hello,bag,world
- Output: bag,hello,without,world

In [ ]:
# Write your code here


## Program 77
Write a program that accepts a sequence of whitespace separated words as input and prints the words after removing all duplicate words and sorting them alphanumerically.

**Example:**
- Input: hello world and practice makes perfect and hello world again
- Output: again and hello makes perfect practice world

In [ ]:
# Write your code here


## Program 79
Write a program that accepts a sentence and calculates the number of letters and digits.

**Example:**
- Input: hello world! 123
- Output: LETTERS 10, DIGITS 3

In [ ]:
# Write your code here


## Program 80
A website requires users to input a username and password to register. Write a program to check the validity of password input by users. Criteria:

1. At least 1 letter between [a-z]
2. At least 1 number between [0-9]
3. At least 1 letter between [A-Z]
4. At least 1 character from [$#@]
5. Minimum length: 6
6. Maximum length: 12

Your program should accept comma separated passwords and print those that match all criteria.

**Example:**
- Input: ABd1234@1,a F1#,2w3E*,2We3345
- Output: ABd1234@1

In [ ]:
# Write your code here


## Program 81
Define a class with a generator which can iterate the numbers, which are divisible by 7, between a given range 0 and n.

In [ ]:
# Write your code here


## Program 82
Write a program to compute the frequency of the words from the input. The output should be sorted alphanumerically.

**Example:**
- Input: New to Python or choosing between Python 2 and Python 3? Read Python 2 or Python 3.
- Output: word:frequency pairs sorted alphanumerically

In [ ]:
# Write your code here


## Program 83
Define a class Person and its two child classes: Male and Female. All classes have a method "getGender" which can print "Male" for Male class and "Female" for Female class.

In [ ]:
# Write your code here


## Program 84
Write a program to generate all sentences where subject is in ["I", "You"] and verb is in ["Play", "Love"] and the object is in ["Hockey", "Football"].

In [ ]:
# Write your code here


## Program 85
Write a program to compress and decompress the string "hello world!hello world!hello world!hello world!".

In [ ]:
# Write your code here


## Program 86
Write a binary search function which searches an item in a sorted list. The function should return the index of the element to be searched in the list.

In [ ]:
# Write your code here


## Program 87
Write a program using generator to print the numbers which can be divisible by 5 and 7 between 0 and n in comma separated form while n is input by console.

**Example:**
- Input: 100
- Output: 0,35,70

In [ ]:
# Write your code here


## Program 88
Write a program using generator to print the even numbers between 0 and n in comma separated form while n is input by console.

**Example:**
- Input: 10
- Output: 0,2,4,6,8,10

In [ ]:
# Write your code here


## Program 89
Write a program using list comprehension to print the Fibonacci Sequence in comma separated form with a given n.

f(n) = 0 if n=0, f(n) = 1 if n=1, f(n) = f(n-1) + f(n-2) if n>1

**Example:**
- Input: 8
- Output: 0,1,1,2,3,5,8,13

In [ ]:
# Write your code here


## Program 90
Assuming email addresses in the "username@companyname.com" format, write a program to print the username of a given email address. Both usernames and company names are composed of letters only.

**Example:**
- Input: john@google.com
- Output: john

In [ ]:
# Write your code here


## Program 91
Define a class named Shape and its subclass Square. The Square class has an init function which takes a length as argument. Both classes have an area function which can print the area of the shape where Shape's area is 0 by default.

In [ ]:
# Write your code here


## Program 92
Write a function that stutters a word as if someone is struggling to read it. The first two letters are repeated twice with an ellipsis ... and space after each, and then the word is pronounced with a question mark ?.

**Examples:**
- stutter("incredible") = "in... in... incredible?"
- stutter("enthusiastic") = "en... en... enthusiastic?"
- stutter("outstanding") = "ou... ou... outstanding?"

Hint: Assume all input is in lower case and at least two characters long.

In [ ]:
# Write your code here


## Program 93
Create a function that takes an angle in radians and returns the corresponding angle in degrees rounded to one decimal place.

**Examples:**
- radians_to_degrees(1) = 57.3
- radians_to_degrees(20) = 1145.9
- radians_to_degrees(50) = 2864.8

In [ ]:
# Write your code here


## Program 94
Establish if a given integer num is a Curzon number. If 1 plus 2 elevated to num is exactly divisible by 1 plus 2 multiplied by num, then num is a Curzon number.

**Condition:** (2^n + 1) is divisible by (2n + 1)

**Examples:**
- is_curzon(5) = True (2⁵ + 1 = 33, 2×5 + 1 = 11, 33 is divisible by 11)
- is_curzon(10) = False
- is_curzon(14) = True

In [ ]:
# Write your code here


## Program 95
Given the side length x, find the area of a hexagon.

**Examples:**
- area_of_hexagon(1) = 2.6
- area_of_hexagon(2) = 10.4
- area_of_hexagon(3) = 23.4

In [ ]:
# Write your code here


## Program 96
Create a function that returns a base-2 (binary) representation of a base-10 (decimal) number.

**Examples:**
- binary(1) = "1"
- binary(5) = "101"
- binary(10) = "1010"

In [ ]:
# Write your code here


## Program 97
Create a function that takes three arguments a, b, c and returns the sum of the numbers that are evenly divided by c from the range a, b inclusive.

**Examples:**
- evenly_divisible(1, 10, 20) = 0
- evenly_divisible(1, 10, 2) = 30
- evenly_divisible(1, 10, 3) = 18

In [ ]:
# Write your code here


## Program 98
Create a function that returns True if a given inequality expression is correct and False otherwise.

**Examples:**
- correct_signs("3 < 7 < 11") = True
- correct_signs("13 > 44 > 33 < 1") = False
- correct_signs("1 < 2 < 6 < 9 > 3") = True

In [ ]:
# Write your code here


## Program 99
Create a function that replaces all the vowels in a string with a specified character.

**Examples:**
- replace_vowels("the aardvark", "#") = "th# ##rdv#rk"
- replace_vowels("minnie mouse", "?") = "m?nn?? m??s?"
- replace_vowels("shakespeare", "*") = "sh*k*sp**r*"

In [ ]:
# Write your code here


## Program 100
Write a function that calculates the factorial of a number recursively.

**Examples:**
- factorial(5) = 120
- factorial(3) = 6
- factorial(1) = 1
- factorial(0) = 1

In [ ]:
# Write your code here


## Program 101
Hamming distance is the number of characters that differ between two strings.

Create a function that computes the hamming distance between two strings.

**Examples:**
- hamming_distance("abcde", "bcdef") = 5
- hamming_distance("abcde", "abcde") = 0
- hamming_distance("strong", "strung") = 1

In [ ]:
# Write your code here


## Program 102
Create a function that takes a list of non-negative integers and strings and returns a new list without the strings.

**Examples:**
- filter_list([1, 2, "a", "b"]) = [1, 2]
- filter_list([1, "a", "b", 0, 15]) = [1, 0, 15]
- filter_list([1, 2, "aasf", "1", "123", 123]) = [1, 2, 123]

In [ ]:
# Write your code here


## Program 103
The "Reverser" takes a string as input and returns that string in reverse order, with the opposite case.

**Examples:**
- reverse("Hello World") = "DLROw OLLEh"
- reverse("ReVeRsE") = "eSrEvEr"
- reverse("Radar") = "RADAr"

In [ ]:
# Write your code here


## Program 104
Unpack the list `writeyourcodehere = [1, 2, 3, 4, 5, 6]` into three variables using destructuring assignment:
- first → 1
- middle → [2, 3, 4, 5]
- last → 6

In [ ]:
writeyourcodehere = [1, 2, 3, 4, 5, 6]
# Unpack the list into first, middle, last


## Program 105
Write a function that calculates the factorial of a number recursively.

**Examples:**
- factorial(5) = 120
- factorial(3) = 6
- factorial(1) = 1
- factorial(0) = 1

In [ ]:
# Write your code here


## Program 106
Write a function that moves all elements of one type to the end of the list.

**Examples:**
- move_to_end([1, 3, 2, 4, 4, 1], 1) = [3, 2, 4, 4, 1, 1]
- move_to_end([7, 8, 9, 1, 2, 3, 4], 9) = [7, 8, 1, 2, 3, 4, 9]
- move_to_end(["a", "a", "a", "b"], "a") = ["b", "a", "a", "a"]

In [ ]:
# Write your code here


## Program 107
Create a function that takes a string and returns a string in which each character is repeated once.

**Examples:**
- double_char("String") = "SSttrriinngg"
- double_char("Hello World!") = "HHeelllloo WWoorrlldd!!"
- double_char("1234! ") = "11223344!! "

In [ ]:
# Write your code here


## Program 108
Create a function that reverses a boolean value and returns the string "boolean expected" if another variable type is given.

**Examples:**
- reverse(True) = False
- reverse(False) = True
- reverse(0) = "boolean expected"
- reverse(None) = "boolean expected"

In [ ]:
# Write your code here


## Program 109
Create a function that returns the thickness (in meters) of a piece of paper after folding it n number of times. The paper starts off with a thickness of 0.5mm.

**Examples:**
- num_layers(1) = "0.001m"
- num_layers(4) = "0.008m"
- num_layers(21) = "1048.576m"

In [ ]:
# Write your code here


## Program 110
Create a function that takes a single string as argument and returns an ordered list containing the indices of all capital letters in the string.

**Examples:**
- index_of_caps("eDaBiT") = [1, 3, 5]
- index_of_caps("eQuINoX") = [1, 3, 4, 6]
- index_of_caps("determine") = []
- index_of_caps("STRIKE") = [0, 1, 2, 3, 4, 5]

In [ ]:
# Write your code here


## Program 111
Using list comprehensions, create a function that finds all even numbers from 1 to the given number.

**Examples:**
- find_even_nums(8) = [2, 4, 6, 8]
- find_even_nums(4) = [2, 4]
- find_even_nums(2) = [2]

In [ ]:
# Write your code here


## Program 112
Create a function that takes a list of strings and integers, and filters out the list so that it returns a list of integers only.

**Examples:**
- filter_list([1, 2, 3, "a", "b", 4]) = [1, 2, 3, 4]
- filter_list(["A", 0, "Edabit", 1729, "Python", 1729]) = [0, 1729]
- filter_list(["Nothing", "here"]) = []

In [ ]:
# Write your code here


## Program 113
Given a list of numbers, create a function which returns the list but with each element's index in the list added to itself (add 0 to index 0, add 1 to index 1, etc.)

**Examples:**
- add_indexes([0, 0, 0, 0, 0]) = [0, 1, 2, 3, 4]
- add_indexes([1, 2, 3, 4, 5]) = [1, 3, 5, 7, 9]
- add_indexes([5, 4, 3, 2, 1]) = [5, 5, 5, 5, 5]

In [ ]:
# Write your code here


## Program 114
Create a function that takes the height and radius of a cone as arguments and returns the volume of the cone rounded to the nearest hundredth.

Formula: V = (1/3) × π × r² × h

**Examples:**
- cone_volume(3, 2) = 12.57
- cone_volume(15, 6) = 565.49
- cone_volume(18, 0) = 0

In [ ]:
# Write your code here


## Program 115
The Triangular Number Sequence: 1, 3, 6, 10, 15, ...

Write a function that gives the number of dots with its corresponding triangle number of the sequence.

**Examples:**
- triangle(1) = 1
- triangle(6) = 21
- triangle(215) = 23220

In [ ]:
# Write your code here


## Program 116
Create a function that takes a list of numbers between 1 and 10 (excluding one number) and returns the missing number.

**Examples:**
- missing_num([1, 2, 3, 4, 6, 7, 8, 9, 10]) = 5
- missing_num([7, 2, 3, 6, 5, 9, 1, 4, 8]) = 10
- missing_num([10, 5, 1, 2, 4, 6, 8, 3, 9]) = 7

In [ ]:
# Write your code here


## Program 117
Write a function that takes a list and a number as arguments. Add the number to the end of the list, then remove the first element of the list. Return the updated list.

**Examples:**
- next_in_line([5, 6, 7, 8, 9], 1) = [6, 7, 8, 9, 1]
- next_in_line([7, 6, 3, 23, 17], 10) = [6, 3, 23, 17, 10]
- next_in_line([], 6) = "No list has been selected"

In [ ]:
# Write your code here


## Program 118
Create the function that takes a list of dictionaries and returns the sum of people's budgets.

**Example:**
```python
get_budgets([
    { 'name': 'John', 'age': 21, 'budget': 23000 },
    { 'name': 'Steve', 'age': 32, 'budget': 40000 },
    { 'name': 'Martin', 'age': 16, 'budget': 2700 }
]) = 65700
```

In [ ]:
# Write your code here


## Program 119
Create a function that takes a string and returns a string with its letters in alphabetical order.

**Examples:**
- alphabet_soup("hello") = "ehllo"
- alphabet_soup("edabit") = "abdeit"
- alphabet_soup("hacker") = "acehkr"

In [ ]:
# Write your code here


## Program 120
Create a function that accepts the principal p, the term in years t, the interest rate r, and the number of compounding periods per year n. The function returns the value at the end of term rounded to the nearest cent.

Formula: A = p × (1 + r/n)^(n × t)

**Examples:**
- compound_interest(10000, 10, 0.06, 12) = 18193.97
- compound_interest(100, 1, 0.05, 1) = 105.0

In [ ]:
# Write your code here


## Program 121
Write a function that takes a list of elements and returns only the integers.

**Examples:**
- return_only_integer([9, 2, "space", "car", "lion", 16]) = [9, 2, 16]
- return_only_integer(["hello", 81, "basketball", 123, "fox"]) = [81, 123]
- return_only_integer(["String", True, 3.3, 1]) = [1]

In [ ]:
# Write your code here


## Program 122
Create a function that takes three parameters where x is the start (inclusive), y is the end (inclusive), and n is the divisor. Return an ordered list with numbers in the range divisible by n.

**Examples:**
- list_operation(1, 10, 3) = [3, 6, 9]
- list_operation(7, 9, 2) = [8]
- list_operation(15, 20, 7) = []

In [ ]:
# Write your code here


## Program 123
Create a function that takes in two lists and returns True if the second list follows the first list by one element (i.e., the second list is the first list shifted to the right by 1).

**Examples:**
- simon_says([1, 2], [5, 1]) = True
- simon_says([1, 2], [5, 5]) = False
- simon_says([1, 2, 3, 4, 5], [0, 1, 2, 3, 4]) = True

In [ ]:
# Write your code here


## Program 124
A group of friends have decided to start a secret society. The name will be the first letter of each of their names, sorted in alphabetical order.

**Examples:**
- society_name(["Adam", "Sarah", "Malcolm"]) = "AMS"
- society_name(["Harry", "Newt", "Luna", "Cho"]) = "CHLN"

In [ ]:
# Write your code here


## Program 125
An isogram is a word that has no duplicate letters. Create a function that takes a string and returns either True or False depending on whether or not it's an isogram.

**Examples:**
- is_isogram("Algorism") = True
- is_isogram("PasSword") = False
- is_isogram("Consecutive") = False

Note: Not case sensitive.

In [ ]:
# Write your code here


## Program 126
Create a function that takes a string and returns True or False, depending on whether the characters are in order or not.

**Examples:**
- is_in_order("abc") = True
- is_in_order("edabit") = False
- is_in_order("123") = True
- is_in_order("xyzz") = True

In [ ]:
# Write your code here


## Program 127
Create a function that takes a number as an argument and returns True or False depending on whether the number is symmetrical or not. A number is symmetrical when it is the same as its reverse.

**Examples:**
- is_symmetrical(7227) = True
- is_symmetrical(12567) = False
- is_symmetrical(44444444) = True
- is_symmetrical(1112111) = True

In [ ]:
# Write your code here


## Program 128
Given a string of numbers separated by a comma and space, return the product of the numbers.

**Examples:**
- multiply_nums("2, 3") = 6
- multiply_nums("1, 2, 3, 4") = 24
- multiply_nums("54, 75, 453, 0") = 0
- multiply_nums("10, -2") = -20

In [ ]:
# Write your code here


## Program 129
Create a function that squares every digit of a number.

**Examples:**
- square_digits(9119) = 811181
- square_digits(2483) = 416649
- square_digits(3212) = 9414

Note: The function receives an integer and must return an integer.

In [ ]:
# Write your code here


## Program 130
Create a function that sorts a list and removes all duplicate items from it.

**Examples:**
- setify([1, 3, 3, 5, 5]) = [1, 3, 5]
- setify([4, 4, 4, 4]) = [4]
- setify([5, 7, 8, 9, 10, 15]) = [5, 7, 8, 9, 10, 15]
- setify([3, 3, 3, 2, 1]) = [1, 2, 3]

In [ ]:
# Write your code here


## Program 131
Create a function that returns the mean of all digits.

**Examples:**
- mean(42) = 3
- mean(12345) = 3
- mean(666) = 6

Note: mean = sum of digits / number of digits (result is always an integer)

In [ ]:
# Write your code here


## Program 132
Create a function that takes an integer and returns a list from 1 to the given number, where:
1. If the number can be divided evenly by 4, amplify it by 10 (return 10× the number)
2. If the number cannot be divided evenly by 4, simply return the number

**Examples:**
- amplify(4) = [1, 2, 3, 40]
- amplify(3) = [1, 2, 3]
- amplify(25) = [1, 2, 3, 40, 5, 6, 7, 80, ...]

In [ ]:
# Write your code here


## Program 133
Create a function that takes a list of numbers and returns the number that's unique.

**Examples:**
- unique([3, 3, 3, 7, 3, 3]) = 7
- unique([0, 0, 0.77, 0, 0]) = 0.77
- unique([0, 1, 1, 1, 1, 1, 1, 1]) = 0

Note: There is always exactly one unique number.

In [ ]:
# Write your code here


## Program 134
Create a Circle class with a radius provided by an argument. The class must have two methods: getArea() (π×r²) and getPerimeter() (2π×r). Round results to the nearest integer.

**Examples:**
- circy = Circle(11); circy.getArea() → 380
- circy = Circle(4.44); circy.getPerimeter() → 28

In [ ]:
# Write your code here


## Program 135
Create a function that takes a list of strings and returns a list, sorted from shortest to longest.

**Examples:**
- sort_by_length(["Google", "Apple", "Microsoft"]) = ["Apple", "Google", "Microsoft"]
- sort_by_length(["Turing", "Einstein", "Jung"]) = ["Jung", "Turing", "Einstein"]

In [ ]:
# Write your code here


## Program 136
Create a function that validates whether three given integers form a Pythagorean triplet. The sum of the squares of the two smallest integers must equal the square of the largest.

**Examples:**
- is_triplet(3, 4, 5) = True (3² + 4² = 25 = 5²)
- is_triplet(13, 5, 12) = True
- is_triplet(1, 2, 3) = False

Note: Numbers may not be given in sorted order.

In [ ]:
# Write your code here


## Program 137
Create a function that takes three integer arguments (a, b, c) and returns the amount of integers which are of equal value.

**Examples:**
- equal(3, 4, 3) = 2
- equal(1, 1, 1) = 3
- equal(3, 4, 1) = 0

Note: Your function must return 0, 2 or 3.

In [ ]:
# Write your code here


## Program 138
Write a function that converts a dictionary into a list of keys-values tuples, sorted alphabetically.

**Examples:**
- dict_to_list({"D": 1, "B": 2, "C": 3}) = [("B", 2), ("C", 3), ("D", 1)]
- dict_to_list({"likes": 2, "dislikes": 3, "followers": 10}) = [("dislikes", 3), ("followers", 10), ("likes", 2)]

In [ ]:
# Write your code here


## Program 139
Write a function that creates a dictionary with each (key, value) pair being the (lower case, upper case) versions of a letter, respectively.

**Examples:**
- mapping(["p", "s"]) = {"p": "P", "s": "S"}
- mapping(["a", "b", "c"]) = {"a": "A", "b": "B", "c": "C"}

Note: All letters in the input list will always be lowercase.

In [ ]:
# Write your code here


## Program 140
Write a function that replaces all vowels in a string with a specified vowel.

**Examples:**
- vow_replace("apples and bananas", "u") = "upplus und bununus"
- vow_replace("cheese casserole", "o") = "chooso cossorolo"
- vow_replace("stuffed jalapeno poppers", "e") = "steffed jelepene peppers"

Note: All words will be lowercase. Y is not considered a vowel.

In [ ]:
# Write your code here


## Program 141
Create a function that takes a string as input and capitalizes a letter if its ASCII code is even and returns its lower case version if its ASCII code is odd.

**Examples:**
- ascii_capitalize("to be or not to be!") = "To Be oR NoT To Be!"
- ascii_capitalize("THE LITTLE MERMAID") = "THe LiTTLe meRmaiD"
- ascii_capitalize("Oh what a beautiful morning.") = "oH wHaT a BeauTiFuL moRNiNg."

In [ ]:
# Write your code here
